rules/no-array-for-each.js

```diff
@@ -175,13 +180,35 @@ function getFixFunction(callExpression, functionInfo, context) {
 			return false;
 		}
 
-		if (callback.body.type !== 'BlockStatement') {
+		if (callback.body.type !== 'BlockStatement' && !isOptionalChaining) {
```

Comment:
Can you explain why we need this here?

Reply:
2 tests would fail. `foo?.forEach((element) => console.log(element));` and `foo?.forEach(element => console.log(element));`

```
    Actual:␊
    if (foo) {␊
    	for (const element of foo) console.log(element)␊
    };␊
    ␊
    Expected:␊
    if (foo) {␊
    	for (const element of foo) console.log(element)␊
    }`,
```

Reply:
```js
if (foo) {
  for (const element of foo) console.log(element);
}
```

This is expected, try to fix.